### PR TITLE
Validate augment inputs with configuration

### PR DESCRIPTION
**Pull request by anonx3247**
## Description

Created a set of unit tests on the `augmentInputsWithConfiguration` in `front/lib/actions/mcp_internal_actions/input_configuration.ts` To validate the correct augmentation.

## Tests

Here is the list of tests:
```
basic functionality > should return original inputs when no schema properties exist
basic functionality > should return original inputs when no missing required properties
DATA_SOURCE mime type > should augment inputs with data source configuration
DATA_SOURCE mime type > should handle data source with filter instead of sId
TABLE mime type > should augment inputs with table configuration
AGENT mime type > should augment inputs with agent configuration
AGENT mime type > should throw error when childAgentId is missing
REASONING_MODEL mime type > should augment inputs with reasoning model configuration
NULLABLE_TIME_FRAME mime type > should augment inputs with time frame configuration
NULLABLE_TIME_FRAME mime type > should return null when timeFrame is not configured
JSON_SCHEMA mime type > should augment inputs with JSON schema configuration
JSON_SCHEMA mime type > should return null when jsonSchema is not configured
primitive types > should augment inputs with string configuration
primitive types > should augment inputs with number configuration
primitive types > should augment inputs with boolean configuration
primitive types > should augment inputs with enum configuration
primitive types > should augment inputs with list configuration
DUST_APP mime type > should augment inputs with Dust app configuration
DUST_APP mime type > should throw error when dustAppConfiguration is missing
error handling > should throw error for invalid string type
error handling > should throw error for invalid number type
error handling > should throw error for invalid list type
nested objects and complex schemas > should handle nested objects with configurable properties
nested objects and complex schemas > should handle multiple missing properties
edge cases > should handle empty data sources array
edge cases > should handle null data sources
edge cases > should preserve existing inputs and only add missing ones
```

Also ran and passed all other tests.

## Note *BUG FOUND*

While trying to validate the `JSON_SCHEMA mime type > should augment inputs with JSON schema configuration
` test I had bugs where AJV couldn't compile the regular JSON Schema given by Zod (`JsonSchemaSchema` which is a Zod schema of a json schema). It turned out that AJV cannot handle nested `$refs` so by changing a parameter in Zod's options all `$ref` definitions appear in the root of the schema. 

```ts
-  Object.entries(ConfigurableToolInputSchemas).map(([key, schema]) => [
-    key,
-    zodToJsonSchema(schema),
-  ])
+  Object.entries(ConfigurableToolInputSchemas).map(([key, schema]) => {
+    const jsonSchema = zodToJsonSchema(schema, {
+      // Use 'none' to inline all references instead of creating $ref pointers
+      $refStrategy: 'none',
+    });
+    // Remove $schema property since these are property definitions, not standalone schemas
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    const { $schema, ...schemaWithoutDollarSchema } = jsonSchema;
+    return [key, schemaWithoutDollarSchema];
+  })
 ) as Omit<
   Record<InternalToolInputMimeType, JSONSchema>,
   typeof INTERNAL_MIME_TYPES.TOOL_INPUT.ENUM
   ```

## Risk

Close to none, only real change was to a buggy part of the Zod Json Schema generator to remove a bug it had. Apart from that the rest of the PR is tests which don't impact the codebase further.

## Deploy Plan

Deploy front

### CODE REVIEW

File: front/lib/actions/mcp_internal_actions/input_configuration.test.ts [hunk 1]
```diff
@@ -0,0 +1,1078 @@
...
+        rawInputs,
+        actionConfiguration: config,
+      });
+
+      expect(result).toEqual({
+        dataSource: [],
+      });
+    });
+
+    it("should preserve existing inputs and only add missing ones", () => {
```
Comment (Fraggle) [inline front/lib/actions/mcp_internal_actions/input_configuration.test.ts:1032]: not really an edge case :)

**Review by Fraggle — APPROVED** [review]
Can you add at least one test to make sure that additional configuration that is no longer needed in the inputSchema are ignored ? (this one will easily break after refactoring how we augment).

Context:
- rawInput has some values A, B
- inputSchema requires A, B, C
- addtionalConfiguration has C and D (that was in an "previous" version of the input schema)

Expection:

A, B and C only

**Comment by anonx3247** [comment]
I accidentally clicked an auto-merge before your comments, I'll make the new changes on the same branch and make a new PR that I'll link to here.

**Comment by Fraggle** [comment]
ah, I see you reverted (note: it was okay to merge as is and just add another PR for the extra test(s)) #trust
